### PR TITLE
fix: logged errors are reported to error reporting

### PR DESCRIPTION
**Pull request by DominicKramer**
Fixes: #121

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -69,18 +68,26 @@ function getCurrentTraceFromAgent() {
...
     this.stackdriverLog = logging(options).log(this.logName, {
       removeCircular: true,
     });
+
+    // serviceContext.service is required by the Error Reporting
+    // API.  Without it, errors that are logged with level 'error'
+    // or higher will not be displayed in the Error Reporting
+    // console
+    if (!this.serviceContext.service) {
+      this.serviceContext.service = 'default';
```
Comment (ofrobots) [inline src/index.ts]: I left a comment earlier but I don't see it anywhere.

If the user *intended* to provide the service context, but it incorrectly, we should throw.

Comment (ofrobots) [reply src/index.ts]: The other comment I had added was.. is there precent from other languages?

**Comment by ofrobots** [comment]
We talked in person about the next steps
1. Throw on bad config provided by users
2. Implement auto-detection of service context when running on Google Cloud.

I think the former should be addressed in this PR, and the latter be implemented as a follow-on feature.

**Comment by DominicKramer** [comment]
@ofrobots I've updated this PR with the features we discussed.  I've opened issue #135 for the follow-on work.  PTAL

**Comment by DominicKramer** [comment]
@ofrobots and I discussed this PR in person and, after updating this PR to not modify the default section of the README, this PR is ready to land.

**Pull request merged**